src/providers/analytics/provider.ts

```diff
@@ -34,49 +44,64 @@ export class IndexAnalyticsProvider implements AnalyticsProvider {
       provider,
       coingeckoService,
     )
     const navProvider = new IndexNavProvider(provider, coingeckoService)
-    const supplyProvider = new IndexSupplyProvider(provider)
-    // For now we can save time by always assuming Ethereum
-    const chainId = 1
-    // Token data fetched statically to avoid lengthy contract interactions
-    const token = TokenData[address.toLowerCase()]
-    const totalSupply = await supplyProvider.getSupply(address)
-    const coingeckoRes = await coingeckoService.getTokenPrice({
+
+    const supplyPromise = options.includeTotalSupply
+      ? new IndexSupplyProvider(provider).getSupply(address)
+      : Promise.resolve(null)
+    const marketCapPromise = marketCapProvider.getMarketCap(address)
+    const navPricePromise = navProvider.getNav(address)
+    const coingeckoPromise = coingeckoService.getTokenPrice({
       address,
       chainId,
       baseCurrency,
       include24hrChange: true,
-      include24hrVol: true,
+      include24hrVol: options.include24hrVolume ?? false,
     })
-    const coingeckoData = coingeckoRes[address.toLowerCase()]
-    const change24hLabel = CoinGeckoUtils.get24hChangeLabel(baseCurrency)
-    const change24h =
-      coingeckoData === undefined ? 0 : coingeckoData[change24hLabel]
-    const marketPrice =
-      coingeckoData === undefined ? 0 : coingeckoData[baseCurrency]
-    const volume24hLabel = CoinGeckoUtils.get24hVolumeLabel(baseCurrency)
-    const volume24h =
-      coingeckoData === undefined ? 0 : coingeckoData[volume24hLabel]
-    const marketCap = await marketCapProvider.getMarketCap(address)
-    const navPrice = await navProvider.getNav(address)
-    const supplyFormatted = utils.formatUnits(totalSupply.toString())
+
+    const [totalSupply, marketCap, navPrice, coingeckoRes] =
+      await Promise.allSettled([
+        supplyPromise,
+        marketCapPromise,
+        navPricePromise,
+        coingeckoPromise,
+      ])
+    const coingeckoData =
+      coingeckoRes.status === "fulfilled"
+        ? coingeckoRes.value?.[address.toLowerCase()]
+        : null
+    const token = TokenData[address.toLowerCase()]
+
     return {
       symbol: token.symbol,
       address,
       name: token.name,
       decimals: token.decimals,
-      totalSupply: supplyFormatted,
-      marketPrice,
-      navPrice,
-      marketCap,
-      change24h,
-      volume24h,
+      totalSupply:
+        options.includeTotalSupply && totalSupply
+          ? utils.formatUnits(totalSupply.toString())
+          : null,
+      marketPrice: coingeckoData?.[baseCurrency] ?? null,
+      navPrice: navPrice.status === "fulfilled" ? navPrice.value : null,
+      marketCap: marketCap.status === "fulfilled" ? marketCap.value : null,
+      change24h:
+        coingeckoData?.[CoinGeckoUtils.get24hChangeLabel(baseCurrency)] ?? null,
+      volume24h: options.include24hrVolume
+        ? coingeckoData?.[CoinGeckoUtils.get24hVolumeLabel(baseCurrency)] ??
+          null
+        : null,
```

Comment:
For readability and easier debugging could you make those `const`s again?

Reply:
Maybe also a small util function like `getValueOrNull()`?

Reply:
Updated to use const for label, but didn't add the util as it started making the typescript too messy.

Reply:
I don't see it how it could make it messier. Was thinking about something like this:

```typescript
function getValueOrNull(result: any): any | null {
  return result.status === "fulfilled" ? result.value : null
}
```